_pages/home/release-notes/checkout-libraries-release-notes.md

```diff
@@ -0,0 +1,395 @@
+    return this.createDeliveryAddressCommand.execute(address);
+  }
+
+  protected createDeliveryAddressCommand: Command<Address, unknown> =
+  this.commandService.create<Address>(
+    (payload) =>
+      this.checkoutPreconditions().pipe(
+        switchMap(([userId, cartId]) => {
+          return this.checkoutDeliveryAddressConnector
+            .createAddress(userId, cartId, payload)
+            .pipe(
+              tap(() => {
+                if (userId !== OCC_USER_ID_ANONYMOUS) {
+                  this.eventService.dispatch(
+                    { userId },
+                    LoadUserAddressesEvent
+                  );
+                }
+              }),
+              map((address) => {
+                address.titleCode = payload.titleCode;
+                if (payload.region?.isocodeShort) {
+                  address.region = {
+                    ...address.region,
+                    isocodeShort: payload.region.isocodeShort,
+                  };
+                }
+                return address;
+              }),
+              tap((address) =>
+                this.eventService.dispatch(
+                  {
+                    userId,
+                    cartId,
+                    address,
+                  },
+                  CheckoutDeliveryAddressCreatedEvent
+                )
+              )
+            );
+        })
+      ),
+    {
+      strategy: CommandStrategy.CancelPrevious,
+    }
+  );
+```
+
+1. Commands can dispatch events (for other side effects)
+
+The following snippet is taken from the createDeliveryAddressCommand snipet above. After successfully creating an address, we have dispatched two [events](https://sap.github.io/spartacus-docs/event-service/#dispatching-individual-events) under the tap operator in order to be caught by our event listeners to perform additional actions.
+
+```typescript
+return this.checkoutDeliveryAddressConnector
+  .createAddress(userId, cartId, payload)
+  .pipe(
+    tap(() => {
+      if (userId !== OCC_USER_ID_ANONYMOUS) {
+        this.eventService.dispatch( <-- 1st dispatched event
+          { userId },
+          LoadUserAddressesEvent
+        );
+      }
+    }),
+    map((address) => {
+      address.titleCode = payload.titleCode;
+      if (payload.region?.isocodeShort) {
+        address.region = {
+          ...address.region,
+          isocodeShort: payload.region.isocodeShort,
+        };
+      }
+      return address;
+    }),
+    tap((address) =>
+      this.eventService.dispatch( <-- 2nd dispatched event
+        {
+          userId,
+          cartId,
+          address,
+        },
+        CheckoutDeliveryAddressCreatedEvent
+      )
+    )
+```
+
+After an event that was dispatched during a command or query, our event listener that listened for [CheckoutDeliveryAddressCreatedEvent](#checkoutdeliveryaddresscreatedevent) fires a global message to let the user know that they have successfully created the delivery address.
```

Comment:
should we really remove this? It is a reference to the event explained under the events section, where it will automatically position the reader to the event

Reply:
I think so, because we should remove the list of events from this doc.
You can maybe link a file using the permalink here, or link to the event docs (which is not yet updated, but maybe put a todo for Matt).

Reply:
I removed the TODO that references to put a reference for the event docs because we already have it in the generalized event section.

Moreover, the events there are automatically generated if I recall correctly